uaclient/cli.py

```diff
@@ -839,21 +856,105 @@ def action_config_set(args, *, cfg, **kwargs):
             validate_url = util.PROXY_VALIDATION_APT_HTTP_URL
         else:
             validate_url = util.PROXY_VALIDATION_APT_HTTPS_URL
         util.validate_proxy(protocol_type, set_value, validate_url)
-        kwargs = {
-            "http_proxy": cfg.apt_http_proxy,
-            "https_proxy": cfg.apt_https_proxy,
-        }
-        kwargs[set_key[4:]] = set_value
+
+        if proxy_scope == "apt":
+            event.info(ua_status.MESSAGE_WARNING_APT_PROXY_SETUP)
+            proxy_scope = "global"  # change the global_apt_* proxy config
+            set_key = "global_" + set_key
+            protocol_type_location = 2
+
+        # unset the current proxy setting
+        # either global or ua scoped
+        unset_current = False
+        if proxy_scope == "global":  # check if ua-scoped proxy set
+            unset_current = (
+                True
+                if getattr(cfg, "ua_apt_http_proxy")
+                or getattr(cfg, "ua_apt_https_proxy")
+                else False
+            )
+            if unset_current:
+                prev_val = "ua_apt_" + protocol_type + "_proxy"
+                event.info(
+                    ua_status.MESSAGE_WARNING_APT_PROXY_OVERWRITE.format(
+                        proxy=prev_val
+                    )
+                )
+        elif proxy_scope == "ua":  # check if global-scoped proxy set
+            unset_current = (
+                True
+                if getattr(cfg, "global_apt_http_proxy")
+                or getattr(cfg, "global_apt_https_proxy")
+                else False
+            )
+            if not unset_current:
+                unset_current = (
+                    True
+                    if getattr(cfg, "apt_http_proxy")
+                    or getattr(cfg, "apt_https_proxy")
+                    else False
+                )
+            if unset_current:
+                prev_val = "global_apt_" + protocol_type + "_proxy"
+                event.info(
+                    ua_status.MESSAGE_WARNING_APT_PROXY_OVERWRITE.format(
+                        proxy=prev_val
+                    )
+                )
+
+        if unset_current:
+            kwargs = {
+                "http_proxy": None,
+                "https_proxy": None,
+                "proxy_scope": cfg.apt_proxy_scope,
+            }
+            setup_apt_proxy(**kwargs)
+
+        if proxy_scope == "global":
+            kwargs = {
+                "http_proxy": cfg.global_apt_http_proxy,
+                "https_proxy": cfg.global_apt_https_proxy,
+                "proxy_scope": cfg.apt_proxy_scope,
+            }
+        elif proxy_scope == "ua":
+            kwargs = {
+                "http_proxy": cfg.ua_apt_http_proxy,
+                "https_proxy": cfg.ua_apt_https_proxy,
+                "proxy_scope": cfg.apt_proxy_scope,
+            }
+
+        key_groups = set_key.split("_")
+        key = "_".join(key_groups[protocol_type_location:])
+        kwargs[key] = set_value
```

Comment:
I know this is an adaptation of what was already here, but I think this would be easier to read if we just did this:
```suggestion
        if "https" in set_key:
            kwargs["https_proxy"] = set_value
        else:
            kwargs["http_proxy"] = set_value
```